File: Python Codes/Palindrome.py
num = int(input("Enter a number: "))
original_num = num
rev = 0
while num>0:
    dig = num%10
    rev = rev*10 + dig
    num //= 10
if rev==original_num:
    print("Palindrome")
else:
    print("Not a palindrome")

File: C program/New Batch/Palindrome.cpp
#include<stdio.h>
int main(){
    int num;//1234
    printf("Enter a number: ");
    scanf("%d",&num);
    int n = num;//121
    int rev=0;//4321
    while(num>0){
        rev = rev*10 + (num%10);
        num /= 10;
    }
    printf("Reverse: %d\n",rev);
    if(rev==n){
        printf("\nPalindrome");
    }
    else{
        printf("\nNon Palindrome");
    }
    return 0;
}